src/DataProtection/DataProtection/src/SP800_108/ManagedSP800_108_CTR_HMACSHA512.cs

```diff
@@ -55,6 +57,217 @@ public static void DeriveKeys(byte[] kdk, ArraySegment<byte> label, ArraySegment
         }
     }
 
+#if NET10_0_OR_GREATER
```

Comment:
.NET already has SP800108 in the box. We should probably just use the one that is built-in to .NET if it is available.

https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.sp800108hmaccounterkdf

Reply:
I dont think the API is having the same input \ output args, and I cant understand if I can use the in-box version for the code purposes: destination is being broken apart into `operationSubKey` and `validationSubKey` - could I just slice the `destination` span to get the same data?

sorry, not knowing much about cryptography so can be asking something weird :)

Reply:
Ooh that looks great. Since it has netstandard and netfx targets we could remove a bunch of code in this repo. Should figure out the `operationSubKey`/`validationSubKey` question you have.

Reply:
I have tested against runtime provided SP800108, and perf is slower. I will probably raise an issue for that in dotnet/runtime in some time.

For now keeping this usage